pkg/m3admin/commmon.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package m3admin
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+
+	retryhttp "github.com/hashicorp/go-retryablehttp"
+)
+
+// FormatHostname provides a formatted URL for HTTP reqeusts
+func FormatHostname(domain string) string {
+	return fmt.Sprintf("http://%s.%s:%d", M3coordinatorServiceName, domain, M3coordinatorPort)
+}
+
+// DoHttpReqeust is a simple helper for HTTP requests
+func DoHttpRequest(
```

Comment:
can you get away with this being a non-exported func?

Reply:
Not sure how since this a function that is shared between two different interfaces.